src/Microsoft.AspNet.Mvc.TagHelpers/LinkTagHelper.cs

```diff
@@ -323,26 +324,27 @@ private void BuildFallbackBlock(TagHelperContent builder)
                     }
                 }
 
-                builder.Append(Environment.NewLine);
+                builder.Append(HtmlString.NewLine);
 
                 // Build the <meta /> tag that's used to test for the presence of the stylesheet
-                builder.AppendFormat(
-                    CultureInfo.InvariantCulture,
-                    "<meta name=\"x-stylesheet-fallback-test\" class=\"{0}\" />",
-                    HtmlEncoder.HtmlEncode(FallbackTestClass));
+                builder
+                    .AppendEncoded("<meta name=\"x-stylesheet-fallback-test\" class=\"")
+                    .Append(FallbackTestClass)
```

Comment:
Worth checking for remaining `HtmlEncoder.HtmlEncode()` calls (outside `HtmlHelper.Encode()` and similar) in your branch's product code. You may have fixed all instances of #3122 in this PR.